### ISSUE TITLE
Train on rail car dataset

### ISSUE THREAD

**Issue by lewfish** [azavea/raster-vision#161]

**Comment by lewfish** [comment]
Here are some results for experiments I ran on the rail car dataset. For these experiments, I used two projects: Rail Car 071 (id `43397307-227e-46ab-968a-00382f2b42f5`) and Rail Car 066 (id `b09a6ca4-ca46-4877-a146-33d0055adf58`).

`jm-rail-car-071`
Train on 071 with default anchor scales (which control the "prior" on size of objects) and predict on the same project. It should do well since we're training and testing on the same project. The predictions were made with the `merge_thresh` set higher than usual to 0.5 so that it will only merge together boxes that are overlapping by greater than 0.5 IOU. (This parameter controls the aggregate_predictions process). The NMS being done on each individual chip is also reasonably high at 0.6. I thought that 0.5 was high enough so that overlapping (but not duplicate) detections would not be merged, but I should have run the experiment with a threshold of 1.0 to turn off the merging process. 

It has trouble distinguishing between cars.
![rf-rail-car-071-071](https://user-images.githubusercontent.com/1896461/32907535-5373f25a-cace-11e7-97e2-2455ef763ad9.png)

But does pretty well in some other areas.
![rf-rail-car-071-071-good](https://user-images.githubusercontent.com/1896461/32907685-e5e90c2e-cace-11e7-8615-62d0f5041cda.png)

Despite the training project having 1248 cars, it does not generalize well to a different project. This is probably because cars within a project look similar and have similar backgrounds and are at similar angles. The angle problem could be improved with some better data augmentation.
![rf-rail-car-071-061](https://user-images.githubusercontent.com/1896461/32907895-cf3e7418-cacf-11e7-9b0b-63676243e9ef.png)

`jm-rail-car-071-scale`
Train on 071 with reduced anchor sizes (ranging from 0.05 to 0.5 instead of 0.2 to 0.95) to hopefully better handle small objects. For prediction, I used the above parameter settings, but also set the score threshold to 0.1 instead of 0.5 to increase the recall. As expected the number of false positives went up a lot, but it didn't seem to improve the ability to detect small objects.
![rf-rail-car-071-scale-071](https://user-images.githubusercontent.com/1896461/32908721-68b55182-cad2-11e7-95f8-8df8c51d854b.png)
